### DISCUSSION TITLE
Collection.removeAll doesn't remove entities and just makes them orphan

### DISCUSSION THREAD

**Discussion by sergeyampo** [mikro-orm/mikro-orm#3008] [Q&A] [answered]
Why does Collection.removeAll and Collection.remove method of example relation Book -> BookFiles: 1 -> M generates SQL as:
`UPDATE book_file SET book_id = NULL WHERE book_id = ...`
I expected DELETE operation, so what's the point of such soft deleting or detaching entities?
Also, it's not documented.

**Comment by B4nan** [comment, marked as answer]
Removing item from collection does not mean you delete the target entity, it means you are disconnecting the relation - removing items from collection, not removing entities from database - `Collection.remove()` is not the same as `em.remove()`. If you want to remove it, you need `orphanRemoval: true`, which tells the ORM you don't want orphaned entities to exist, so we know those should be removed. Or call `em.remove()` instead.

**Reply by sergeyampo** [reply]
Can you add this [here](https://mikro-orm.io/api/core/class/ArrayCollection#remove) please?
